policyengine_us/variables/gov/states/mo/tax/income/subtractions/mo_qualified_health_insurance_premiums.py

```diff
@@ -8,55 +8,59 @@ class mo_qualified_health_insurance_premiums(Variable):
         tax_unit = person.tax_unit
 
-        # total_health_insurance_premiums is also a primary input to the MO side of calculation, MO Form 5695, Line 8
+        # total_health_insurance_premiums is also a primary input to
+        # the MO side of calculation, MO Form 5695, Line 8
         # Federal Schedule A, Line 4
         med_expense_deduction = add(
             tax_unit, period, ["medical_expense_deduction"]
         )
 
-        # the ratio of federal medical expense deduction to total medical expenses (out of pocket + premiums)
-        # need division because med_dental_out_of_pocket is in federal tax, but no MO tax
-        # this ratio is then used to scale the health_insurance_premium amount that can be claimed
+        # the ratio of federal medical expense deduction to
+        # total medical expenses (out of pocket + premiums)
+        # need division because med_dental_out_of_pocket is
+        # in federal tax, but no MO tax
+        # this ratio is then used to scale the health_insurance_premium
+        # amount that can be claimed
         # IRS Schedule A Line 1
         tax_unit_health_expenses = add(
             tax_unit,
             period,
             ["medical_out_of_pocket_expenses", "health_insurance_premiums"],
         )
-        # Apply a where statement to avoid division by zero.
-        med_expense_deducted_ratio = where(
-            tax_unit_health_expenses > 0,
-            med_expense_deduction / tax_unit_health_expenses,
-            0,
+        med_expense_ratio = np.zeros_like(tax_unit_health_expenses)
+        mask = tax_unit_health_expenses > 0
+        med_expense_ratio[mask] = (
+            med_expense_deduction[mask] / tax_unit_health_expenses[mask]
         )
 
         person_premiums = person("health_insurance_premiums", period)
         tax_unit_premiums = tax_unit.sum(person_premiums)
 
-        # Line 13 of MO Form 5695, represents the portion of medical expenses already deducted via federal tax itemization
-        deducted_portion = tax_unit_premiums * med_expense_deducted_ratio
+        # Line 13 of MO Form 5695, represents the portion of
+        # medical expenses already deducted via federal tax itemization
+        deducted_portion = tax_unit_premiums * med_expense_ratio
         # subtracts the portion of premiums already deducted from federal tax
         itemized_premiums_amount = tax_unit_premiums - deducted_portion
         itemizes = tax_unit("tax_unit_itemizes", period)
         # Cap at federal taxable income.
         taxable_income = tax_unit("taxable_income", period)
         # Allocate proportionally across people in the tax unit based on share
-        # of premiums. Use where statement to avoid dividing by zero.
-        person_share_of_tax_unit_premiums = where(
-            tax_unit_premiums > 0, person_premiums / tax_unit_premiums, 0
-        )
+        # of premiums.
+        person_share = np.zeros_like(tax_unit_premiums)
```

Comment:
```suggestion
        # Use a mask rather than where to avoid a divide-by-zero warning. Default to zero.
        person_share = np.zeros_like(tax_unit_premiums)
```